### PR TITLE
Add support for unix domain sockets

### PR DESCRIPTION
**Pull request by wilmveel**
fixes: #149

### CODE REVIEW

**Comment by hayd** [comment]
Does this mean the whole library will need --unstable?

**Comment by wilmveel** [comment]
This means you only need unstable when you are using the socket.

File: connection_params.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import { parseDsn } from "./utils.ts";
+import { parseDsn, validHostname } from "./utils.ts";
+import validate = WebAssembly.validate;
```
Comment (hayd) [inline connection_params.ts:2]: unused?

File: tests/connection_params.ts [hunk 2]
```diff
@@ -36,34 +36,34 @@ function withNotAllowedEnv(fn: () => void) {
 
 test("dsnStyleParameters", function () {
   const p = createParams(
-    "postgres://some_user@some_host:10101/deno_postgres",
+    "postgres://some_user@some-host:10101/deno_postgres",
```
Comment (hayd) [inline tests/connection_params.ts:39]: so the underscore in hostname is not valid?

**Comment by hayd** [comment]
Great, thanks! One thing I am a little concerned about is running the tests requiring --unstable, it would be nice to be able to test the lib works without --unstable still (so nothing in --unstable slips into the main logic).

**Comment by wilmveel** [comment]
I got your point. I've changed the approach. 

`ConnectionParams` has a now a function conn which returns a `Promise<Denno.Conn>`. `ConnectionParams` can now direct be used as an input for `Client`. 

As a user wants to use the socket it can be done by providing a connection function. The --unstable flag is only needed on application level when the user provides the connection based on the socket

**Pull request closed without merging**